TAGradingServer/account/account-numerictext-gradeable.php

```diff
@@ -21,6 +21,52 @@
     }
 }
 
+if(User::$user_group == 1){
+    $csv_button = "<label>Upload CSV (WARNING! Previously entered data may be overwritten!): Do not include a header row. Format CSV using one column for student id and one column for each field. Columns and field types must match.</br><input type=\"file\" id=\"csvUpload\" accept=\".csv, .txt\" onchange=\"csvUpload()\"></label>";
+    $csv_upload_functions = "
+        function csvUpload(){
+            var f = $('#csvUpload').get(0).files[0];
+            
+            if(f){
+                var reader = new FileReader();
+                reader.readAsText(f);
+                reader.onload = function(evt) {
+                  parseCsv(reader.result);
+                  
+                }
+                reader.onerror = function(evt){
+                    console.error(\"nope\");
+                }
+            } 
+        }
+        
+        function parseCsv(csv){
+            url = \"{$BASE_URL}/account/ajax/account-numerictext-gradeable.php?course={$_GET['course']}&semester={$_GET['semester']}&g_id={$_GET['g_id']}\";
+            var lines = csv.split(/\\r\\n|\\n/);
+            console.log(lines);
+            console.log(url);
+            $.ajax({
+                type:\"POST\",
+                url:url,
+                data: {
+                    csrf_token: '{$_SESSION['csrf']}',
+                    parsedCsv: lines,
+                    action:\"csv\"
+                },
+                success: function(data, text){
+                    location.reload();
+                },
+                error: function(request, status, error){
+                    window.alert(\"An error has occurred. Contact an administrator.\");
+                }
+            });
+        }";
```

Comment:
So I was talking about [Heredoc](http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc)

In that it allows you to do:
```
$a = <<<HTML
Hello "{$variable}", we're using double quotes and variables and no need for escaping things!
Woo!
HTML;
```